### PR TITLE
Add routing to create the App

### LINKED ISSUE #7: Add Home page
**Description:**

The user sees her/his name in the page header with a welcome text. In the header the user finds a page navigation. The home page shows a calendar with the date of the day as default. User can add a new diary entry by clicking on the add icon. The page also displays a calendar with a monthly view. Days with diary entries are highlighted and the user can access the date by clicking on it. It's also possible to change the month in calendar with drop menu.

**Screenshot:**

image

**Tasks:**

- [x] Add a header component
- [x] Add a page navigation
- [x] Set the header to display the name set on the Welcome page
- [x] Add a card component with a display of the current date and an onClick icon
- [x] Add a calendar component with monthly view and a drop menu to change the month
- [x] Add a function to highlight dates with diary entries, this needs a fetch from MongoDB

**Acceptance criteria:**

- none: if nothing is selected, the card displays the current date and calendar shows the current month

### LINKED ISSUE #22: Routing
Create an App by routing

### PR DESCRIPTION
**Pull request by riittagerhards**
Add routing for the App and install the new version of npm.
Routing from calendar to single entry tbd.

part of #10 & #11
fixes #7
fixes #22

### CODE REVIEW

**Review by welschoff — COMMENTED** [review]
Good Job, Riitta :)

File: src/app/pages/SingleEntry/SingleEntry.tsx [hunk 1]
```diff
@@ -1,27 +1,54 @@
...
-  src,
-  entryTitle,
-  text,
-}: SingleEntryProps): JSX.Element {
+function SingleEntry(): JSX.Element {
+  const { date } = useParams();
+  const [entry, setEntry] = useState<SingleEntryProps | null>();
+
+  const getEntries = async () => {
+    const response = await fetch(`/api/entries/${date}/`);
```
Comment (leonardpaulus) [inline src/app/pages/SingleEntry/SingleEntry.tsx:19]:
```suggestion
    const response = await fetch(`/api/entries/${date}`);
```

File: src/app/pages/SingleEntry/SingleEntry.tsx [hunk 2]
```diff
@@ -1,27 +1,54 @@
...
+      {entry && <Navigation headerTitle={getDate(entry.date)} />}
       <div className={styles.container}>
-        <img className={styles.image} src={src} alt="my picture of the day" />
-        <h1>{entryTitle}</h1>
-        <p className={styles.text}>{text}</p>
+        {entry && (
+          <img
+            className={styles.image}
+            src={entry.imageUrl}
+            alt="my picture of the day"
```
Comment (leonardpaulus) [inline src/app/pages/SingleEntry/SingleEntry.tsx:47]: Would suggest a more specific name or delete it, if it should be more common :)
'My picture of the day' wouldn't give any hint of what can be seen there. :(

**Review by leonardpaulus — APPROVED** [review]
Looks good Riitta! :) 🚀 
Just two small hints :)

**Pull request merged**